dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java

```diff
@@ -309,13 +309,12 @@ private void validateSearchOutsideCaptureScopeParameters(
         validateSearchableAttributes(params, searchableAttributeIds);
       }
 
-      int maxTeLimit = getMaxTeLimit(params);
-      params.setMaxTeLimit(maxTeLimit);
-      if (maxTeLimit > 0
-          && trackedEntityStore.getTrackedEntityCountWithMaxTrackedEntityLimit(params)
-              > maxTeLimit) {
-        throw new IllegalQueryException("maxteicountreached");
+      if (!params.hasTrackedEntities()) {
+        validateMinAttributesToSearch(params);
       }
+
+      params.setMaxTeLimit(getMaxTeLimit(params));
+      validateMaxTrackedEntitiesInResponse(params);
```

Comment:
The method name is not clear enough in my opinion. 

Some suggestions:
validateTrackedEntitySearchResultsWithinLimit
validateMaxTrackedEntitySearchResults
validateSearchResultsWithinMaxTeLimit
validateTotalResultsWithinMaxTeLimit

or some combination of these words. 

"InResponse" sounds misleading since it could indicate the paginated response.